### PR TITLE
Stop parenthesizing context managers on Python 3.9

### PR DESCRIPTION
**Pull request by A5rocks**
### Description

Discovered in https://github.com/python-trio/trio/pull/3106#discussion_r1794256324, black is parenthesizing context managers for Python 3.9 despite that only officially landing in 3.10.

This isn't actually much of an issue as CPython 3.9 allows it and I assume PyPy does too, but it could technically be an issue and does mess things up with `-X oldparser` on 3.9.

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [ ] Add new / update outdated documentation?

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
This feels not worth it. We'd cause churn on projects that already run on 3.9 only. I'd rather stick with the new syntax only at this point.

**Comment by A5rocks** [comment]
> This feels not worth it. We'd cause churn on projects that already run on 3.9 only. I'd rather stick with the new syntax only at this point.

Makes sense. The way it's implemented it doesn't actually downgrade already parenthesized context managers, if I remember correctly. Though that's obviously a bug that could maybe be treated as a feature.

File: src/black/mode.py [hunk 1]
```diff
@@ -1,296 +1,295 @@
...
-            str(int(self.is_pyi)),
-            str(int(self.is_ipynb)),
-            str(int(self.skip_source_first_line)),
-            str(int(self.magic_trailing_comma)),
-            str(int(self.preview)),
-            str(int(self.unstable)),
-            features_and_magics,
-        ]
-        return ".".join(parts)
+"""Data structures configuring Black behavior.
```
[the remark this answers is not in the document]
Comment (A5rocks) [reply src/black/mode.py:1]: Oh, it's probably yet another artifact of me running Windows. I only noticed it happening for the changelog, though I guess it happened here too: for the changelog prettier was writing everything out with CRLF and I had to manually revert that...

**Comment by JelleZijlstra** [comment]
Thanks, but I'm going to reject this, especially since we've already released this style for a long time. Parenthesized context managers realistically work fine in Python 3.9; it's unlikely that anybody is actively using `-X oldparser`.

**Pull request closed without merging**